src/osd/ReplicatedBackend.cc

```diff
@@ -2213,19 +2213,13 @@ bool ReplicatedBackend::handle_push_reply(pg_shard_t peer, PushReplyOp &op, Push
     } else {
       // done!
       get_parent()->on_peer_recover(
-	peer, soid, pi->recovery_info,
-	pi->stat);
-
-      object_stat_sum_t stat;
-      stat.num_bytes_recovered = pi->recovery_info.size;
-      stat.num_keys_recovered = reply->omap_entries.size();
-      stat.num_objects_recovered = 1;
+	peer, soid, pi->recovery_info);
 
       get_parent()->release_locks(pi->lock_manager);
+      object_stat_sum_t stat = pi->stat;
```

Comment:
`pi`  is erased from `pushing` and set to NULL before the if branch. See the two lines following.
The `*_recovered` fields are set in `build_push_op`.